plugin.onedrive/addon.py

```diff
@@ -307,29 +300,22 @@ def export_folder(name, item_id, driveid, destination_folder, base_folder, direc
         extension = utils.Utils.get_extension(f['name']);
         name = utils.Utils.ascii(f['name'])
         if is_folder:
-            export_folder(name, f['id'], driveid, parent_folder, base_folder)
+            export_folder(name, item_driveid, f['id'], driveid, parent_folder, base_folder)
         elif (('video' in f or extension in ext_videos) and content_type == 'video') or ('audio' in f and content_type == 'audio'):
-            params = {'action':'play', 'content_type': content_type, 'item_id': f['id'], 'driveid': driveid}
+            params = {'action':'play', 'content_type': content_type, 'item_driveid': item_driveid, 'item_id': f['id'], 'driveid': driveid}
             url = base_url + '?' + urllib.urlencode(params)
-            with open(os.path.join(parent_folder, name + '.strm'), 'wb') as fo:
-                fo.write(url)
+            f = xbmcvfs.File(parent_folder + name + '.strm', 'w')
+            f.write(url)
+            f.close()
         onedrive.exporting_count += 1
         p = int(onedrive.exporting_count/float(onedrive.exporting_target)*100)
         if onedrive.exporting_percent < p:
             onedrive.exporting_percent = p
-        file_path = os.path.join(parent_folder, name)
+        file_path = parent_folder + name
```

Comment:
please use os.path.join here. It is a bad practice to concatenate strings for folder paths